src/containers/SearchPage/SearchPage.js

```diff
@@ -0,0 +1,62 @@
+import PaginationBar from "../PaginationBar/PaginationBar";
+import Item from "../productItem/Item";
+import "./SearchPage.css";
+
+function SearchPage({ Products, itemsPerPage=20 }) {
+  const { isLoading } = Products;
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+  const getParamValueFromKey=(searchKey)=>{
+    const params = new URLSearchParams(window.location.search)
+    for (const [key, value] of params) {
+      if(key===searchKey)
+      return value;
+    }
+  };
```

Comment:
React Router provides a `useLocation` hook, so we don't have to use `window.location`, this hook tracks the value for changes so our component is updated and avoid bugs, in this case this solves the search bug.  
```suggestion
 const {search} = useLocation()
  if (isLoading) {
    return <h1>Loading...</h1>;
  }
  const getParamValueFromKey=(searchKey)=>{
    const params = new URLSearchParams(search)
    for (const [key, value] of params) {
      if(key===searchKey)
      return value;
    }
  };
```